tutorials/packaging/snap-a-website.md

```diff
@@ -0,0 +1,681 @@
+      - libnss3
+      - libxss1
+      - fontconfig-config
+      - gconf2
+      - libasound2
+      - pulseaudio
+
+    after:
+      - desktop-glib-only
+```
+
+The `after` field indicates that our "`electron-app`" part will only be built "after" the: `desktop-glib-only` part.
+
+#### Build and post-build instructions
+
+Then we need to provide the following information:
+
+* **The command (or commands) to build the app**:
+  Since we are using electron-builder and simply calling `electron-builder` in the project will pick up details from our `package.json` file and generate an executable.
+
+* **What to do with the result of our build**:
+  The result of running `electron-builder` is a `app/dist/linux-unpacked/`, this is what we want in the snap, in a dedicated `app/` directory for consistency.
+
+* **What to package (and in our case, what not to package)**:
+  To ensure the snap package doesn't contain unnecessary files, we tell snapcraft to trim what it pulls into the snap and exclude `app/node_modules/`, which only contains build dependencies.
+
+The following fields are taking care of each step:
+
+```yaml
+    build: node_modules/.bin/electron-builder
+    install: |
+      mkdir $SNAPCRAFT_PART_INSTALL/app
+      mv dist/linux-unpacked/* $SNAPCRAFT_PART_INSTALL/app
+    prime:
+      - -node_modules
+```
+
+And can be added at the end of our part.
+
+Here is our `snapcraft.yaml` at this stage of the process:
+
+```yaml
+parts:
+  electron-app:
+    source: app/
+    plugin: nodejs
+
+    stage-packages:
+      - libnotify4
+      - libappindicator1
+      - libxtst6
+      - libnss3
+      - libxss1
+      - fontconfig-config
+      - gconf2
+      - libasound2
+      - pulseaudio
+
+    after:
+      - desktop-glib-only
+
+    build: node_modules/.bin/electron-builder
+    install: |
+      mkdir $SNAPCRAFT_PART_INSTALL/app
+      mv dist/linux-unpacked/* $SNAPCRAFT_PART_INSTALL/app
+    prime:
+      - -node_modules
+```
+
+### Let's recap
+
+We have added general metadata to ensure our app appears correctly in stores and parts to ensure it bundles our executable and its dependencies in a confined package. But we are still missing one very important bit: the launcher.
+
+## Launcher and desktop integration
+
+To launch the app, the snap needs to know which command to run. This is done within an `apps` section in the `snapcraft.yaml`, where we are going to declare: the name of the app, the command to run and which access we want the app to be granted outside of the snap confinement.
+
+Positive
+: **About confinement**
+By default, a snap will be fully confined and will have no way to interact with other pieces of the system, such as accessing the network, the display and sound servers, etc. As a result, we need to declare explicitly what our app needs access to.
+
+This is what our `apps` section needs to look like.
+
+```yaml
+apps:
+  castlearena:
+    command: env TMPDIR=$XDG_RUNTIME_DIR desktop-launch $SNAP/app/castlearena
```

Comment:
This XDG_RUNTIME_DIR should probably be explained. I have no idea what it does there.